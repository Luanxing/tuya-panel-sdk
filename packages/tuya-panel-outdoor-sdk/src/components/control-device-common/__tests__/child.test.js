/**
 * @jest-environment jsdom
 */
import React from 'react';
import { shallow } from 'enzyme';
import { Utils } from 'tuya-panel-kit';
import Child from '../child';

const { convertX: cx } = Utils.RatioUtils;

describe('child components', () => {
  it('child render', () => {
    const wrapper = shallow(
      <Child
        icon=""
        color="red"
        size={cx(24)}
        iconStyle={{
          width: cx(50),
          height: cx(50),
          borderRadius: cx(50),
          alignItems: 'center',
          justifyContent: 'center',
        }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
