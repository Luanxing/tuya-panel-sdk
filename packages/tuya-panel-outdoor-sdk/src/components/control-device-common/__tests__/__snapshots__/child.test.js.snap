// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`child components child render 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "borderRadius": 100,
      "height": 100,
      "justifyContent": "center",
      "width": 100,
    }
  }
>
  <Component
    color="red"
    d=""
    size={48}
  />
</View>
`;
